### PR TITLE
chore(Authenticator): refactor Router, add RouteContainer

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes

Refactor `Router` and add `RouteContainer`:
- consolidate HTML `View` components from `Authenticator` and `Router` components in `RouteContainer`
- wrap "route" components (e.g. `ConfirmSignUp`) with `RouteContainer`
- fix circular dependency issues in `Authenticator` by removing `defaultComponents` from `CustomComponents` initialization
- add/consolidate/update some types

#### Issue #, if available
NA

#### Description of how you validated changes
- ran integ tests
- manual functionality testing
- created snapshots tests in sample app
- checked imported types in sample app

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/Authenticator/Authenticator.tsx [hunk 1]
```diff
@@ -1,86 +1,128 @@
...
     </Provider>
   );
-}
-
-Authenticator.Provider = Provider;
-Authenticator.ResetPassword = ResetPassword;
-Authenticator.SetupTOTP = SetupTOTP;
-Authenticator.SignIn = SignIn;
-Authenticator.SignUp = SignUp;
-Authenticator.ForceNewPassword = ForceNewPassword;
```
Comment (calebpollman) [inline packages/react/src/components/Authenticator/Authenticator.tsx:86]: Moved the assigning of these components as properties to the primary _index.ts_ that the `Authenticator` is exported from. Something abt the `AuthenticatorContainer` being imported in to this file and wrapping the `Router` along with these property assignments was causing the bundle to get corrupted.

File: packages/react/src/components/Authenticator/shared/SignInSignUpTabs.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ export const SignInSignUpTabs = ({
     <>
       {hideSignUp ? (
         <View data-amplify-router-content="">
-           {route === 'signIn' && <SignIn />}
+          {route === 'signIn' && <SignIn />}
```
Comment (calebpollman) [inline packages/react/src/components/Authenticator/shared/SignInSignUpTabs.tsx:23]: Prettier update (same with ln 35)

File: packages/react/src/components/Authenticator/Router/Router.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    default:
+      console.warn(
+        `Unhandled Authenticator route - please open an issue: ${route}`
+      );
+      return () => null;
+  }
+};
+
+export default function Router({ hideSignUp }: { hideSignUp: boolean }) {
+  const { route } = useAuthenticator();
```
Comment (ErikCH) [inline packages/react/src/components/Authenticator/Router/Router.tsx:48]:
```suggestion
    const { route } = useAuthenticator(({ route }) => [route]);
```

Comment (calebpollman) [reply packages/react/src/components/Authenticator/Router/Router.tsx:48]: Nice catch 👀

**Comment by calebpollman** [comment]
Closing as this branch became somewhat of a testing ground for bundle differences i was seeing between `rollup` and `tsup` and has a lot of needless noise

**Pull request closed without merging**